src/Backups/BackupsWorker.cpp

```diff
@@ -182,12 +178,59 @@ namespace
 }
 
 
-BackupsWorker::BackupsWorker(size_t num_backup_threads, size_t num_restore_threads, bool allow_concurrent_backups_, bool allow_concurrent_restores_)
-    : backups_thread_pool(std::make_unique<ThreadPool>(CurrentMetrics::BackupsThreads, CurrentMetrics::BackupsThreadsActive, num_backup_threads, /* max_free_threads = */ 0, num_backup_threads))
-    , restores_thread_pool(std::make_unique<ThreadPool>(CurrentMetrics::RestoreThreads, CurrentMetrics::RestoreThreadsActive, num_restore_threads, /* max_free_threads = */ 0, num_restore_threads))
+std::unique_ptr<BackupsWorker> BackupsWorker::createFromContext(ContextPtr global_context)
+{
+    const auto & config_ = global_context->getConfigRef();
+    const auto & settings_ = global_context->getSettingsRef();
+
+    BackupsWorker::GlobalSettings backup_restore_settings
+    {
+        .num_backup_threads = config_.getUInt64("backup_threads", settings_.backup_threads),
+        .num_restore_threads = config_.getUInt64("restore_threads", settings_.restore_threads),
+        .allow_concurrent_backups = config_.getBool("backups.allow_concurrent_backups", true),
+        .allow_concurrent_restores = config_.getBool("backups.allow_concurrent_restores", true),
+        .stale_backups_restores_check_period_ms = config_.getUInt64("backups.stale_backups_restores_check_period_ms", 600000), // 10 minutes
+        .stale_backups_restores_cleanup_timeout_ms = config_.getUInt64("backups.stale_backups_restores_cleanup_timeout_ms", 86400000), // 24 hours
+        .consecutive_failed_checks_to_consider_as_stale = config_.getUInt64("backups.consecutive_failed_checks_to_consider_as_stale", 3),
+        .root_zookeeper_path = config_.getString("backups.zookeeper_path", "/clickhouse/backups")
+    };
+
+    auto result = std::make_unique<BackupsWorker>(backup_restore_settings);
+
+    if (global_context->hasZooKeeper())
+    {
```

Comment:
Please add a comment here about why you don't create `backup_restore_cleanup_thread` if there is no ZooKeeper.